.github/workflows/gpu_test.yml

```diff
@@ -0,0 +1,57 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: GPU Test V3
+
+on:
+  push:
+    branches: [ v3 ]
+  pull_request:
+    branches: [ v3 ]
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  test:
+    name: py=${{ matrix.python-version }}, np=${{ matrix.numpy-version }}, deps=${{ matrix.dependency-set }}
+
+    runs-on: gpu-runner
+    strategy:
+      matrix:
+        python-version: ['3.11']
+        numpy-version: ['2.0']
+        dependency-set: ["minimal"]
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: cuda-toolkit
+      uses: Jimver/cuda-toolkit@v0.2.16
+      id: cuda-toolkit
+      with:
+        cuda: '12.5.0'
+      run: |
+        echo "Installed cuda version is: ${{steps.cuda-toolkit.outputs.cuda}}"
+        echo "Cuda install location: ${{steps.cuda-toolkit.outputs.CUDA_PATH}}"
+        nvcc -V
```

Comment:
I'm seeing a github action lint error from this section: 

```
[Error](https://github.com/zarr-developers/zarr-python/actions/runs/10569457903/workflow)
a step cannot have both the `uses` and `run` keys
```

Reply:
Pushed a change that should hopefully fix this. Moved the run portion to the GPU check step instead.